libraries/core/src/test/java/com/paritytrading/philadelphia/FIXConnectionStatus.java

```diff
@@ -15,10 +15,12 @@
  */
 package com.paritytrading.philadelphia;
 
+import java.nio.channels.GatheringByteChannel;
+import java.nio.channels.ReadableByteChannel;
 import java.util.ArrayList;
 import java.util.List;
 
-class FIXConnectionStatus implements FIXConnectionStatusListener {
+class FIXConnectionStatus<CHANNEL extends ReadableByteChannel & GatheringByteChannel> implements FIXConnectionStatusListener<CHANNEL> {
```

Comment:
Do we need to repeat the type parameter bounds for `CHANNEL` here?

Reply:
Yes we do, unless we want to hardcode it to a specific class - there are no typedefs in Java... 

On the other hand if we remove the `Session::getConnection() : FIXConnection`, we can get rid of pretty much all generics. I will raise a separate PR with that change as it will be breaking more client code and I assume it may be contentious, while the current set of changes are mostly backwards compatible.